GTViewController/Classes/Custom Classes/GTFollowupModalView.m

```diff
@@ -115,7 +97,27 @@ - (instancetype)initFromElement:(TBXMLElement *)element withStyle:(GTPageStyle*)
                 currentY += buttonPairView.frame.size.height + 10;
                 
             }else if ([modalComponentElementName isEqual:kName_Thank_You]) {
-                self.thankYouPage = nil;
+                
```

Comment:
Ideally you would have a class or structure that has a property for each attribute type. Each of these classes would take that class as the parameter to it's init method and the init method would then just look at the properties it is interested in. Then you could just create the attribute class once at the top of the method and each if statement could have a one liner in it.